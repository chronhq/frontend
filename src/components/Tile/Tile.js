import React from 'react';
import { Link } from 'react-router-dom';

import './Tile.less';

function isOdd(n) {
  return Math.abs(n % 2) === 1;
}

/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/anchor-is-valid */
const Tile = ({ course, lng }) => (
  <div className={isOdd(course.id) ? 'tile odd' : 'tile'}>
  {/*
  <div className='tile-wrapper'>
*/}
    <div className='tile-left' />
    <div className='tile-center'>
      <Link to={`${course.url}`}>
        <div className='tile__content'>
          <h3 className='tile__title'>
            {' '}
            {course.name[lng]}
            {' '}
          </h3>
          <p className='tile__description'>
            {course.description[lng]}
          </p>
        </div>
      </Link>
    </div>
    <div className='tile-right' />
  </div>
);

export default Tile;
