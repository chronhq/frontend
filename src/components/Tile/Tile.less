@import "../../styles/colors.less";

@base01: #222; //
@base02: #000;
@border-decor: rgba(238, 238, 238, 0.3); // Head decor

@hex--height: 20em;
@hex--height_modile: 12em;
@hex--width: 11em;
@hex--med: 2em;
@hex--mod: 2em;
@hex--color: rgba(255, 255, 255, 0.6);
@hex-color-hover: rgba(255, 255, 255, 0.3);


.tile.odd {
  margin-top: calc(@hex--height*0.5 ~'+ 3px');
  @media screen and (max-width: 725px) {
    margin-top: 0;
  }
}


.tile__content {
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-content: center;
  justify-content: center;
  width: 150%;
  height: 60%;
  transform: translate(-16%, 30%);
  text-align: center;
  z-index: 20;
  & h3 {
    margin: 0;
    font-weight: 400;
    font-size: 1.125rem;
    line-height: 20px;
    color: red;
    text-transform: uppercase;
  }
  & a {
    color: inherit;
    text-decoration: inherit;
  }
  & p {
    font-size: 0.9rem;
    color: white;
    font-style: italic;
    padding: 3px;
    line-height: 1.1em;
    text-align: left;
  }

  @media screen and (max-width: 725px) {
    text-align: left;
    width: 100%;
    height: 100%;
    align-self: center;
    transform: none;
    padding-left: 0.5rem;
  }
}

.tile-left {
  border-right: @hex--width/2 solid @hex--color;
  border-top: @hex--height/2 solid transparent;
  border-bottom: @hex--height/2 solid transparent;
  transition : border 500ms ease-out;
  @media screen and (max-width: 725px) {
    display: none;
  }
}

.tile-center {
  width: @hex--width;
  height: @hex--height;
  background-color: @hex--color;
  transition : background-color 500ms ease-out;
  @media screen and (max-width: 725px) {
    margin-bottom: 10px;
    width: 100%;
    height: @hex--height_modile;
  }
}

.tile-right {
  border-left: @hex--width/2 solid @hex--color;
  border-top: @hex--height/2 solid transparent;
  border-bottom: @hex--height/2 solid transparent;
  transition : border 500ms ease-out;
  @media screen and (max-width: 725px) {
    border-left: @hex--mod solid @hex--color;
    border-bottom: @hex--height_modile/2 solid transparent;
    border-top: @hex--height_modile/2 solid transparent;
  }
}

.tile {
  display: flex;
  flex-direction: row;
  width: @hex--width*2;
  height: @hex--height;
  margin-right: calc(-@hex--height*0.25);
  &:hover {
    .tile-left {
      border-right-color: @hex-color-hover;
    }
    .tile-center {
      background-color: @hex-color-hover;
    }
    .tile-right {
      border-left-color: @hex-color-hover;
    }
  }
  @media screen and (max-width: 725px) {
    margin-bottom: 10px;
    width: 100%;
    height: @hex--height_modile;
  }
}

.tile {
  &:first-child {
    &::before {
      @media screen and (max-width: 725px) {
        border: none;
      }
    }
    h3 {
      color: white;
    }
  }
}

