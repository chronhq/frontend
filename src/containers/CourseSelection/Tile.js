import React from 'react';

const Tile = ({ course, disabled, selectCourse, selected }) => {
  const className = selected ? 'courseTileSelected' : 'courseTileRegular';
  return (
    <div className={className} >
      <button
        disabled={disabled}
        onClick={() => selectCourse(course.id)}
      >{selected && 'SELECTED'}{course.name}</button>
    </div>
  );
};

export default Tile;
