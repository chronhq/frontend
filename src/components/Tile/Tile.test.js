import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Tile from './Tile';

const course = {
  url: {
    ru: 'урл',
    en: 'url'
  },
  name: {
    ru: 'name',
    en: 'name'
  },
  description: {
    ru: 'desc',
    en: 'desc'
  },
};

const TestComponent = () => (
  <Router>
    <Tile course={course} lng='ru' />
  </Router>
);


// it must render
it('Tile render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});

// Snapshot tests
it('Tile Snapshot', () => {
  const tree = renderer
    .create(<TestComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
