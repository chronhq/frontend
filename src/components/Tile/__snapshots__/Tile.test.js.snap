// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile Snapshot 1`] = `
<div
  className="tile"
>
  <div
    className="tile-left"
  />
  <div
    className="tile-center"
  >
    <a
      href="/[object Object]"
      onClick={[Function]}
    >
      <div
        className="tile__content"
      >
        <h3
          className="tile__title"
        >
           
          name
           
        </h3>
        <p
          className="tile__description"
        >
          desc
        </p>
      </div>
    </a>
  </div>
  <div
    className="tile-right"
  />
</div>
`;
